### DISCUSSION TITLE
Adding a comma at the end of lists in JSONs

### DISCUSSION THREAD

**Discussion by NoobTracker** [ScratchAddons/ScratchAddons#146] [Discussion]
Take a look at this: 
```
{
  "name": "Discuss button",
  "description": "Adds the discuss button to the navbar.",
  "userscripts": [
    {
      "url": "userscript.js",
      "matches": ["https://scratch.mit.edu/*"]
    }
  ],
  "options": [
    {
      "name": "Remove tips button",
      "id": "removeIdeasBtn",
      "type": "boolean",
      "default": false
    }
  ],
  "tags": ["community", "forums"],
  "enabled_by_default": false
}
```
As you can see, after `"enabled_by_default": false` there is no comma. That's imo a real problem, since it makes screwing things up _incredibly easy_. For example, when you want to add another boolean, you just copy
```

    {
      "name": "Remove tips button",
      "id": "removeIdeasBtn",
      "type": "boolean",
      "default": false
    }
```
and paste it below the other thing and then edit the parameters. Right? **Nope.** Of course not, I probably wouldn't say that like that if it wasn't a problem. You have to add a comma after the first item. 
So, my suggestion is to add a comma after those lines. It's almost a reflex for me to add a comma after an item because I may want to add something so I better should do that when I create that item and it's pretty annoying for me that you can't have commas at the end of `!byte`s in ASM. 
Imo there's not really a disadvantage in adding those (at least in the future) and it could save a lot of headaches. And one other result of having to add those commas is that GitHub says that you also changed the line above although you just added a comma. Not a huge problem, but still another reason to add them at the end of lists.

Since everything in a repository should have the same style, I think that that should be defined so everyone does the same thing.

**Comment by NoobTracker** [comment]
> 
> 
> No trailing commas. To the hell with that.

What's the reason? I can see that there's a discussion whether you should use tabs or spaces since both options have their advantages, but trailing commas don't have a single disadvantage but some advantages.

Edit: Huh, where's that post I quoted?

**Comment by jeffalo** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas#Trailing_commas_in_JSON:~:text=trailing%20commas%20are%20not%20allowed%20in%20JSON

**Comment by Hans5958** [comment]
I disagree for using trailing commas. I just don't like a single comma hanging in the end, instead of none indicating the end of the object, and I don't want to deal with quirks of it.

Also:

> ## Trailing commas in JSON
> 
> Trailing commas in objects were only introduced in ECMAScript 5. As JSON is based on JavaScript's syntax prior to ES5, trailing commas are not allowed in JSON.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas#Trailing_commas_in_JSON

**Comment by NoobTracker** [comment]
argh why has JSON also a problem with that?!

**Comment by mxmou** [comment]
> argh why has JSON also a problem with that?!

Your initial example is from a JSON file.

**Comment by NoobTracker** [comment]
> 
> 
> > argh why has JSON also a problem with that?!
> 
> Your initial example is from a JSON file.

Yeah by "also" I mean not only ASM.

**Comment by apple502j** [comment]
We use JSON.parse to parse JSON files. JSON.parse does not allow trailing comma. We do not add JSON parsing library just to allow trailing comma.

**Comment by NoobTracker** [comment]
Yeah okay I thought that trailing commas are supported by JSON.

**Comment by WorldLanguages** [comment]
We could convert addons.json to a js module, and then it would support trailing commas.

**Comment by WorldLanguages** [comment]
Oh nevermind, this is about addon.json, not addons.json

**Comment by Hans5958** [comment]
I'd rather avoid that process of overengineering. JSON is already enough for script metadata, okay?
